File: Daily Challanges/2026-05-May/2026-05-09.py
def transpose(matrix):
    m = len(matrix)
    n = len(matrix[0])
    transposed_matrix = []
    for i in range(n):
        temp = []
        for j in range(m):
            temp.append(matrix[j][i])
        transposed_matrix.append(temp)
    return transposed_matrix

print(transpose([[1, 2, 3], [4, 5, 6]]))
print(transpose([[1, 2], [3, 4], [5, 6]]))

File: Daily Challanges/2026-05-May/2026-05-09.js
function transpose(matrix) {
  const m = matrix.length;
  const n = matrix[0].length;
  const transposedMatrix = [];
  for (let i = 0; i < n; i++) {
    const temp = [];
    for (let j = 0; j < m; j++) {
      temp.push(matrix[j][i]);
    }
    transposedMatrix.push(temp);
  }
  return transposedMatrix;
}

console.log(
  transpose([
    [1, 2, 3],
    [4, 5, 6],
  ]),
);
console.log(
  transpose([
    [1, 2],
    [3, 4],
    [5, 6],
  ]),
);
